swatch-billable-usage/src/main/java/com/redhat/swatch/billable/usage/services/BillableUsageService.java

```diff
@@ -243,4 +252,31 @@ private void createRemittance(
         contractCoverage.isGratis() ? BillableUsage.Status.GRATIS : BillableUsage.Status.PENDING);
     usage.setUuid(newRemittance.getUuid());
   }
+
+  private void updateUsageMeter(BillableUsage usage, double contractCoverage, double billable) {
+    if (usage.getProductId() == null
+        || usage.getMetricId() == null
+        || usage.getBillingProvider() == null
+        || usage.getStatus() == null) {
+      return;
+    }
+    List<String> tags =
+        new ArrayList<>(
+            List.of(
+                "product_tag", usage.getProductId(),
+                "metric_id", usage.getMetricId(),
+                "billing_provider", usage.getBillingProvider().value(),
+                "status", usage.getStatus().value()));
+    log.error("This is the code calculated coverage from the contracts: {}", contractCoverage);
+    if (usage.getCurrentTotal() > 0) {
+      double coverage =
+          usage.getCurrentTotal() * usage.getBillingFactor() > contractCoverage
+              ? contractCoverage
+              : usage.getCurrentTotal() * usage.getBillingFactor();
+      meterRegistry.counter(COVERED_USAGE_METRIC, tags.toArray(new String[0])).increment(coverage);
+    }
+    if (billable > 0) {
+      meterRegistry.counter(BILLABLE_USAGE_METRIC, tags.toArray(new String[0])).increment(billable);
```

Comment:
For better readibility, use the meter builder API as done in: https://github.com/RedHatInsights/rhsm-subscriptions/pull/3957/files#diff-3688a052ae8bb8ce56defd1ac8ab4d9a5f90687036d69c3b9a3b1a44940d8ff3R376